chunky/src/java/se/llbit/chunky/renderer/projection/OmniDirectionalStereoProjector.java

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2014 Jesper Öqvist <jesper@llbit.se>
+ *
+ * This file is part of Chunky.
+ *
+ * Chunky is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Chunky is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with Chunky.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.llbit.chunky.renderer.projection;
+
+import java.util.Random;
+
+import org.apache.commons.math3.util.FastMath;
+
+import se.llbit.math.Vector3d;
+
+/**
+ * A projector for Omni-Directional Stereo (ODS) images. x is mapped to yaw, y is mapped to
+ * pitch.
+ * @see <a href="https://developers.google.com/vr/jump/rendering-ods-content.pdf">Rendering Omni‐directional Stereo Content</a>
+ */
+public class OmniDirectionalStereoProjector implements Projector {
+	/**
+	 * The eye that this projector produces rays.
+	 */
+	private final Eye eye;
+	/**
+	 * The interpupillary distance of the viewer, in meters.
+	 */
+	private double interpupillaryDistance = 0.069;
+
+	public OmniDirectionalStereoProjector(Eye eye) {
+		this.eye = eye;
+	}
+
+	@Override
+	public void apply(double x, double y, Random random, Vector3d o,
+			Vector3d d) {
+		apply(x, y, o, d);
+	}
+
+	@Override
+	public void apply(double x, double y, Vector3d o, Vector3d d) {
+		double theta = (x + 0.5) * FastMath.PI - FastMath.PI;
+		double phi = FastMath.PI / 2 - (y + 0.5) * FastMath.PI;
+				
+		double scale;
```

Comment:
The scale parameter can be set when an ODS projector is created, thus avoiding this if-statement in the projection code. The ODS projector is only created once, but projects very many rays.